cmd/kubeadm/app/preflight/checks.go

```diff
@@ -843,13 +850,46 @@ func (ResolveCheck) Check() (warnings, errors []error) {
 	return warnings, errors
 }
 
+// ImagePullCheck will pull container images used by kubeadm
+type ImagePullCheck struct {
+	Images    images.Images
+	ImageList []string
+}
+
+// Name returns the label for ImagePullCheck
+func (ImagePullCheck) Name() string {
+	return "ImagePull"
+}
+
+// Check pulls images required by kubeadm. This is a mutating check
+func (i ImagePullCheck) Check() (warnings, errors []error) {
+	for _, image := range i.ImageList {
+		found, err := i.Images.Exists(image)
+		if found == 1 {
+			continue
+		}
+		if err != nil {
+			errors = append(errors, fmt.Errorf("failed to check if image exists [%s]: %v", image, err))
+			continue
+		}
+		if err := i.Images.Pull(image); err != nil {
```

Comment:
Does this print to STDOUT? I think that would be worthwhile to do, to show progress to the user.
This would be a special preflight check in that sense, something in between.